cpp/models/ode_secir/parameters_io.h

```diff
@@ -499,6 +499,29 @@ IOResult<void> read_input_data(std::vector<Model>& model, Date date, const std::
     return success();
 }
 
+/**
+ * @brief Converts input data from one range of models to another with different type.
+ * 
+ * @tparam FP Floating point type (default: double).
+ * @param[in] model_from VectorRange of Node%s each containing a Model with the input data.
+ * @param[in,out] model_to VectorRange of Node%s each containing a Model to be initialized with data.
+ *
+ * @return An IOResult indicating success or failure.
+ */
+template<class FP>
+void convert_input_data_type(const mio::VectorRange<Node<Model<ScalarType>>>& model_from, const mio::VectorRange<Node<Model<FP>>>& model_to)
```

Comment:
Since you are using const VectorRange instead of ConstVectorRange, do we even need the latter? We probably cannot enforce usage of ConstVectorRange, anyways

Reply:
They should behave differently, as ConstVectorRange uses const_iterators, meaning the values the iterator points to cannot be changed. 
Maybe the 'model_from' should be defined as ConstVectorRange.